composition-go/composition.go

```diff
@@ -41,9 +50,60 @@ var (
 	pool sync.Pool
 )
 
+func updateSchemas(subgraphs []*Subgraph) ([]*Subgraph, error) {
+	updatedSubgraphs := make([]*Subgraph, 0, len(subgraphs))
+	for _, subgraph := range subgraphs {
+		if subgraph.Schema != "" {
+			updatedSubgraphs = append(updatedSubgraphs, subgraph)
+			continue
+		}
+		req, err := http.NewRequest("POST", subgraph.URL, strings.NewReader(sdlQuery))
```

Comment:
How about moving the introspection of a subgraph into a separate function?

Reply:
Good idea, that way we can also simplify the error messages a bit. I've refactor it, PTAL.